common4j/src/main/com/microsoft/identity/common/java/util/ported/Pair.java

```diff
@@ -23,10 +23,54 @@
 
 package com.microsoft.identity.common.java.util.ported;
 
+import java.util.List;
+import java.util.Objects;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 import lombok.AllArgsConstructor;
+import lombok.NonNull;
 
 @AllArgsConstructor
 public class Pair<T, U> {
     public final T first;
     public final U second;
+
+    @Override
+    @SuppressFBWarnings(value = "NP_METHOD_PARAMETER_TIGHTENS_ANNOTATION",
+            justification = "Equals function should be able to take in a null object.")
+    public boolean equals(@Nullable final Object obj) {
+        if (obj == null){
+            return false;
+        }
+
+        if (!(obj instanceof Pair)) {
+            return false;
+        }
+
+        final Pair<?, ?> castedObj = (Pair<?, ?>) obj;
+        return Objects.equals(castedObj.first, first) && Objects.equals(castedObj.second, second);
+    }
+
+    /**
+     * Compute a hash code using the hash codes of the underlying objects
+     *
+     * @return a hashcode of the Pair
+     */
+    @Override
+    public int hashCode() {
+        int result = (first != null ? first.hashCode() : 0);
+        result = 31 * result + (second != null ? second.hashCode() : 0);
+        return result;
+    }
+
+    /**
+     * Add this item to a list if a copy of the same key-value pair doesn't exist.
+     * */
+    public void addToListIfNotExist(@NonNull final List<Pair<T, U>> listToBeAdded) {
+        if (!listToBeAdded.contains(this)) {
+            listToBeAdded.add(this);
+        }
+    }
```

Comment:
Is what we're trying to prevent duplicate k-v-p's or duplicate keys?

Reply:
i.e. - would the server accept ?brkr=2&brkr=1 or ?brkr=1,2?

Reply:
ADFS seems to accept them. eSTS, however, might not (haven't heard back from them yet, but they rejects duplicated brkr=1)